corehq/apps/userreports/templates/userreports/edit_report_config.html

```diff
@@ -5,13 +5,69 @@
+                        <h5 class="modal-title">{% trans 'Are you sure you want to delete this report'%}</h5>
+                        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                          <span aria-hidden="true">&times;</span>
+                        </button>
+                      </div>
+                      <div class="modal-body">
+                        <fieldset>
+                            <legend>{% trans 'Deleting this will break below apps that refer the report' %}</legend>
+                            <ul>
+                                {% for app in referring_apps %}
+                                    <li>
+                                        <a href="{{ app.module_url }}">{{ app.module_name}}</a> module in
+                                        <a href="{{ app.app_url }}">{{app.app_name}}</a> app
+                                    </li>
+                                {% endfor %}
+                            </ul>
+                        </fieldset>
+                      </div>
+                      <form method='post' action="{% url 'delete_configurable_report' domain report.get_id %}" >
+                          {% csrf_token %}
+                      <div class="modal-footer">
+                        <button type="button" class="btn btn-default btn-primary" data-dismiss="modal">
+                          {% trans "Cancel" %}
+                        </button>
+                        <button type="submit" value="{% trans 'Delete Report'%}" class="disable-on-submit btn btn-danger btn-secondary">{% trans 'Delete Report'%}</button>
+                      </div>
+                      </form>
+                    </div>
+                  </div>
+                </div>
+            {% else %}
+                <form method='post' action="{% url 'delete_configurable_report' domain report.get_id %}" >
+                    {% csrf_token %}
+                        <input type="submit" value="{% trans 'Delete Report'%}" class="btn btn-danger disable-on-submit pull-right">
+                </form>
             {% endif %}
-            <a href="{% url 'configurable' domain report.get_id %}" class="btn btn-default">{% trans 'View report' %}</a>
-            <a href="{% url 'configurable_report_json' domain report.get_id %}" class="btn btn-default">{% trans 'Report Source (Advanced)' %}</a>
-        </form>
+        {% endif %}
+        <a href="{% url 'configurable' domain report.get_id %}" class="btn btn-default">{% trans 'View report' %}</a>
+        <a href="{% url 'configurable_report_json' domain report.get_id %}" class="btn btn-default">{% trans 'Report Source (Advanced)' %}</a>
+    {% endif %}
+    <br/>
+    {% if not referring_apps  %}
+        <label class="label label-danger">This report is not referred by any apps in this domain. Please note, a static report might be referred by apps in other domains.</label>
```

Comment:
```suggestion
        <label class="label label-danger">This report is not referred by any current apps in this domain. Please note, reports may be referred to older builds and a static report might be referred by apps in other domains.</label>
```

Reply:
Ha, saw this after I already made similar change.